azure-migrate/migrate-at-scale-vmware-agentles/Agentless VMware automation/AzMigrate_StartTestMigration.ps1

```diff
@@ -88,7 +88,91 @@ Function ProcessItemImpl($processor, $csvItem, $reportItem) {
+    }
+    else {
+        $paramsNIC1.Add("NicId", $UpdatedNIC1ID)
+    }  
+
+    $NIC1_TEST_SUBNET_NAME = $csvItem.TFO_NIC1_TEST_SUBNET_NAME
+    if ([string]::IsNullOrEmpty($NIC1_TEST_SUBNET_NAME)) {
+        $processor.Logger.LogTrace("TFO_NIC1_TEST_SUBNET_NAME is not mentioned for: '$($sourceMachineName)'")
+    }
+    else {
+        $paramsNIC1.Add("TestNicSubnet", $NIC1_TEST_SUBNET_NAME)
+    }
+
+    $processor.Logger.LogTrace("nic1test subnet name is: '$($NIC1_TEST_SUBNET_NAME)' and nic1id is: '$($UpdatedNIC1ID)'")
+
+    # NIC parameters to pass to New-AzMigrateServerReplication
+    $params = @{}
+    $paramsNIC2 = @{}    
+    $UpdatedNIC2ID = $csvItem.TFO_NIC2_ID
+    if ([string]::IsNullOrEmpty($UpdatedNIC2ID)) {
+        $processor.Logger.LogTrace("UPDATED_NIC2_ID is not mentioned for: '$($sourceMachineName)'")
+        if ([string]::IsNullOrEmpty($ReplicatingServermachine.ProviderSpecificDetail.VMNic[1].NicId))
+        {
+            $processor.Logger.LogTrace("We didn't find NicId at the second VMNic in replicating server for: '$($sourceMachineName)'")
+        }
+        else {
+            $processor.Logger.LogTrace("We found NicId at the second VMNic in replicating server, so we are going to use this for: '$($sourceMachineName)'")
+            $paramsNIC2.Add("NicId", $ReplicatingServermachine.ProviderSpecificDetail.VMNic[1].NicId)
+        }
+    }
+    else {
+        $paramsNIC2.Add("NicId", $UpdatedNIC2ID)
+    }
+
+    $NIC2_TEST_SUBNET_NAME = $csvItem.TFO_NIC2_TEST_SUBNET_NAME
+    if ([string]::IsNullOrEmpty($NIC2_TEST_SUBNET_NAME)) {
+        $processor.Logger.LogTrace("TFO_NIC2_TEST_SUBNET_NAME is not mentioned for: '$($sourceMachineName)'")
+    }
+    else {
+        $paramsNIC2.Add("TestNicSubnet", $NIC2_TEST_SUBNET_NAME)
+    }
+
+    #Assumption is that if $UpdatedNIC1ID is not provided then probably it doesnt need to be added
+    # we can also add the below code when we check this for the first time but it will be in a nested fashion so doing it here for simplicity 
+    if (-not ([string]::IsNullOrEmpty($UpdatedNIC1ID) -and [string]::IsNullOrEmpty($NIC1_TEST_SUBNET_NAME) -and [string]::IsNullOrEmpty($ReplicatingServermachine.ProviderSpecificDetail.VMNic[0].NicId))) {
+        $Nic1Mapping = New-AzMigrateTestNicMapping @paramsNIC1
+        if(-not $Nic1Mapping){
+            $processor.Logger.LogTrace("Nic1Mapping is not initialized for: '$($sourceMachineName)'")
+        }
+        else {      
+            $NicMapping += $Nic1Mapping        
+        }
+    }
+    #Assumption is that if $UpdatedNIC2ID is not provided then probably it doesnt need to be added
+    # we can also add the below code when we check this for the first time but it will be in a nested fashion so doing it here for simplicity 
+    if (-not ([string]::IsNullOrEmpty($UpdatedNIC2ID) -and [string]::IsNullOrEmpty($NIC2_TEST_SUBNET_NAME) -and [string]::IsNullOrEmpty($ReplicatingServermachine.ProviderSpecificDetail.VMNic[1].NicId))) {
```

Comment:
Please check the logic. It will enter inside if condition when $UpdatedNIC2ID is not null or empty.